sycl/source/kernel_bundle.cpp

```diff
@@ -115,14 +115,10 @@ bool kernel_bundle_plain::is_specialization_constant_set(
 
 const std::vector<device>
 removeDuplicateDevices(const std::vector<device> &Devs) {
-  auto compareDevices = [](device a, device b) {
-    return getSyclObjImpl(a)->getHandleRef() <
-           getSyclObjImpl(b)->getHandleRef();
-  };
-  std::set<device, decltype(compareDevices)> UniqueDeviceSet(
-      Devs.begin(), Devs.end(), compareDevices);
-  std::vector<device> UniqueDevices(UniqueDeviceSet.begin(),
-                                    UniqueDeviceSet.end());
+  std::vector<device> UniqueDevices(Devs.begin(), Devs.end());
+  auto last = std::unique(UniqueDevices.begin(), UniqueDevices.end());
```

Comment:
This call does not remove all duplicates. It only removes those duplicates that go consecutively. See the example here: https://en.cppreference.com/w/cpp/algorithm/unique
If 'Devs' has {A, B, A}, then the function (as written now) returns {A, B, A}

Reply:
@v-klochkov hm.. you are right, will fix.

Reply:
fixed